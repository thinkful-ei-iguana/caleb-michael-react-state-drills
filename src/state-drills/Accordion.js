import React from "react";

export default class Accordion extends React.Component {
  static defaultProps = {
    sections: [{ title: "Default Title", content: "Default Content" }]
  };

  constructor(props) {
    super(props);

    this.state = {
      lastButton: null
    };
  }

  onClickHandler = e => {
    this.setState({
      lastButton: e.target.value
    });
  };

  render() {
    return (
      <ul>
        {this.props.sections.map((section, index) => {
          return (
            <li key={index}>
              <button type="button" onClick={this.onClickHandler} value={index}>
                {section.title}
              </button>
              {this.state.lastButton === `${index}` && <p>{section.content}</p>}
            </li>
          );
        })}
      </ul>
    );
  }
}

//

// return (
//   <div className='view'>
//     {onStageParticipants.map(function(participant) {
//       return (
//       <div className='user'>
//       <h3 className='name'>{participant.name}</h3>
//       <img className='avatar' src={participant.avatar} alt="avatar"/>
//       </div>
//       )
//   })}
//   </div>
