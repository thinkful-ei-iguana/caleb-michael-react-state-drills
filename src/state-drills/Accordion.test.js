import React from "react";
import { configure } from "enzyme";
import { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Accordion from "./Accordion";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

const testArray = [
  {
    title: "Section 1",
    content: "Lorem ipsum dolor sit amet consectetur adipisicing elit."
  },
  {
    title: "Section 2",
    content:
      "Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!"
  },
  {
    title: "Section 3",
    content:
      "Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?"
  }
];

describe("<Accordion />", () => {
  it("renders this UI with default props, no props sent in", () => {
    const wrapper = shallow(<Accordion />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders no sections as active by default with section props input", () => {
    const wrapper = shallow(<Accordion sections={testArray} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the first section when clicked", () => {
    const wrapper = mount(<Accordion sections={testArray} />);
    wrapper
      .find("button")
      .at(1)
      .simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders only the last section opened after multiple sections have been clicked", () => {
    const wrapper = mount(<Accordion sections={testArray} />);
    console.log(wrapper.find("button"));
    wrapper
      .find("button")
      .at(1)
      .simulate("click");
    wrapper
      .find("button")
      .at(2)
      .simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
