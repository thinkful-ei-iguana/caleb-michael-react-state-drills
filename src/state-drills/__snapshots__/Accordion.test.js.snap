// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders no sections as active by default with section props input 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
      value={0}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
      value={1}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
      value={2}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`<Accordion /> renders this UI with default props, no props sent in 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
      value={0}
    >
      Default Title
    </button>
  </li>
</ul>
`;
